modin/core/storage_formats/pandas/parsers.py

```diff
@@ -653,9 +674,49 @@ def parse(fname, **kwargs):
         columns = [c for c in columns if c not in df.index.names and c in df.columns]
         if columns is not None:
             df = df[columns]
-        # Append the length of the index here to build it externally
         return _split_result_for_readers(0, num_splits, df) + [idx, df.dtypes]
 
+    @staticmethod
+    @doc(
+        _doc_parse_func,
+        parameters="""files_for_parser : list
+    List of files to be read.
+storage_options : dict
+    Parameters for specific storage engine.""",
+    )
+    def parse_fsspec_files(files_for_parser, storage_options, **kwargs):
+        from pyarrow.parquet import ParquetFile
+
+        columns = kwargs.get("columns", None)
+
+        chunks = []
+        for file_for_parser in files_for_parser:
+            if isinstance(file_for_parser.path, IOBase):
+                context = nullcontext(file_for_parser.path)
+            else:
+                context = fsspec.open(file_for_parser.path, **storage_options)
+            with context as f:
+                chunk = (
+                    ParquetFile(f)
+                    .read_row_groups(
+                        list(
+                            range(
+                                file_for_parser.row_group_start,
+                                file_for_parser.row_group_end,
+                            )
+                        ),
+                        columns=columns,
+                        use_pandas_metadata=True,
+                    )
+                    .to_pandas()
+                )
+            columns = [
+                c for c in columns if c not in chunk.index.names and c in chunk.columns
```

Comment:
this reduced `columns` (potentially) for each chunk, and each subsequent chunk receives (potentially) smaller amount of columns, is this desired?

Reply:
Not sure what this does - could someone please clarify? Seems to be removing the columns in the index of the current chunk for subsequent chunks - does that mean that the columns in the index are the ones that are actually materialized in the current chunk, while the ones in the columns list are the ones that are in the other chunks?

Reply:
Yeah I think this was copied over from the previous method. I don't think this is actually safe to keep, but I don't understand why it was here even in the old code.

Reply:
I now do understand why this was kept in the code. Please see: https://github.com/modin-project/modin/issues/1366

This case should now be addressed when we process columns in `_read` instead.